AlphaWallet/Tokens/Helpers/CachedERC1155ContractDictionary.swift

```diff
@@ -0,0 +1,76 @@
+//
+//  CachedERC1155ContractDictionary.swift
+//  AlphaWallet
+//
+//  Created by Jerome Chan on 12/4/22.
+//
+
+import Foundation
+
+class CachedERC1155ContractDictionary {
+    private var baseDictionary: [AlphaWallet.Address: Bool] = [AlphaWallet.Address: Bool]()
+    private var fileUrl: URL
+    private var encoder: JSONEncoder
+
+    static func loadFromCache(fileName: String) -> CachedERC1155ContractDictionary? {
+        do {
+            var url: URL = try FileManager.default.url(for: .cachesDirectory, in: .userDomainMask, appropriateFor: nil, create: true)
+            url.appendPathComponent(fileName)
+            let cache: CachedERC1155ContractDictionary = CachedERC1155ContractDictionary(url: url)
+            if FileManager.default.fileExists(atPath: url.path) {
+                cache.readFromFileUrl()
+            }
+            return cache
+        } catch {
+            return nil
+        }
+    }
+
+    private init(url: URL) {
+        self.fileUrl = url
+        self.encoder = JSONEncoder()
+    }
+
+    func isERC1155Contract(for address: AlphaWallet.Address) -> Bool? {
+        return baseDictionary[address]
+    }
+
+    func setContract(for address: AlphaWallet.Address, _ result: Bool) {
+        baseDictionary[address] = result
+        writeToFileUrl()
+    }
+
+    func remove() {
+        do {
+            try FileManager.default.removeItem(at: fileUrl)
+        } catch {
+            // Do nothing
+            NSLog("CachedERC1155ContractDictionary::remove Exception: \(error)")
+        }
+    }
+
+    private func writeToFileUrl() {
+        do {
+            let data = try encoder.encode(baseDictionary)
+            if let jsonString = String(data: data, encoding: .utf8) {
+                try jsonString.write(to: fileUrl, atomically: true, encoding: .utf8)
+            }
+        } catch {
+            // Do nothing
+            NSLog("CachedERC1155ContractDictionary::writeToFileUrl Exception: \(error)")
```

Comment:
Would you use `verboseLog()` instead? and format like this so it's easier to search and filter off:

```
verboseLog("[CachedERC1155ContractDictionary] writeToFileUrl error: \(error)")
```

Reply:
There's this too:

```
NSLog("CachedERC1155ContractDictionary::remove Exception: \(error)")
```